### PR TITLE
Emulation for fetching stash balance

### PR DESCRIPTION
**Pull request by karthikeyan-cere**

### CODE REVIEW

File: scripts/tokenomics-emulation/emulations/stash-account-balance.emulation.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    private readonly network: Network
+  ) {}
+
+  public async run(): Promise<void> {
+    console.log(`Fetch balance of Validators and Nominators Stash account`);
+    const genericValidatorsCount = this.networkConfig.validators.amount;
+    const nominatorsCount = this.networkConfig.nominators.amount;
+    const genesisValidatorsCount = this.networkConfig.genesis_validators_amount;
+
+    console.log(`About to fetch balance of Genesis Validators stash account - ${genesisValidatorsCount}\n`);
```
Comment (AndreiNavoichyk) [inline scripts/tokenomics-emulation/emulations/stash-account-balance.emulation.ts:17]: Move common code to the private method and reuse it.

File: scripts/tokenomics-emulation/network.ts [hunk 2]
```diff
@@ -81,6 +81,21 @@ class Network {
...
+   * Fetch native token balance
+   * @param address account addess
+   * @returns raw balance
+   */
+  public async getRawBalance(address: string) {
+    console.log(`About to get balance for: ${address}`);
+    const {
+      data: { free: balance },
+    } = await this.api.query.system.account(address);
+    const formatedBalance = (+balance / 10 ** this.decimals).toFixed(10);
```
Comment (AndreiNavoichyk) [inline scripts/tokenomics-emulation/network.ts:95]: What is the last `10`? Move it to the constant with understandable name

Comment (karthikeyan-cere) [reply scripts/tokenomics-emulation/network.ts]: Moved to config.

File: scripts/tokenomics-emulation/emulations/stash-account-balance.emulation.ts (hunk 1 above)
Comment (karthikeyan-cere) [reply scripts/tokenomics-emulation/emulations/stash-account-balance.emulation.ts]: Moved to private method

File: scripts/tokenomics-emulation/config.json [hunk 3]
```diff
@@ -28,6 +28,7 @@
     "cere_sc_address": "5C8VXSSmizvyUoKaTkyYTsqrchj3rAgjNHbNPJANhZkXqSfR",
     "gas_limit": -1,
     "smart_contract_cere_token_amount_default": 0,
+    "format_balance_decimals": 10,
```
Comment (AndreiNavoichyk) [inline scripts/tokenomics-emulation/config.json:31]: You can use decimals which we have. It is no sense to have 2 decimals: 1 for formatting and 1 for chain spec.

File: scripts/tokenomics-emulation/emulations/stash-account-balance.emulation.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    for (let i = 1; i <= nominatorsCount; i++) {
+      await this.processAccount(`nominator-${i}-stash`);
+    }
+  }
+
+  private async processAccount(name: string) {
+    const stashAccount = JSON.parse(
+      fs.readFileSync(`../generate-accounts/accounts/all/${name}`, "utf-8")
+    );
+    const balance = await this.network.getRawBalance(stashAccount.ss58Address, this.networkConfig.format_balance_decimals);
```
Comment (AndreiNavoichyk) [inline scripts/tokenomics-emulation/emulations/stash-account-balance.emulation.ts:43]: Use already existing decimals from the config

**Review by AndreiNavoichyk — APPROVED** [review]

File: scripts/tokenomics-emulation/config.json (hunk 3 above)
Comment (karthikeyan-cere) [reply scripts/tokenomics-emulation/config.json]: using the decimals

File: scripts/tokenomics-emulation/emulations/stash-account-balance.emulation.ts (hunk 4 above)
Comment (karthikeyan-cere) [reply scripts/tokenomics-emulation/emulations/stash-account-balance.emulation.ts]: using decimals

**Review by AndreiNavoichyk — APPROVED** [review]

**Pull request merged**